### ISSUE TITLE
[worldclock] US locale defaults to 24 hour time

### ISSUE THREAD

**Issue by tsimonq2** [lxqt/lxqt#1538]
In the United States, most people consider 24 hour time to be "military time". If our goal is to Americanize some things, we might want to switch on 12 hour by default.

It's easy enough to do in my distribution configuration, but I'd like to have the discussion upstream about possible other solutions for this.

**Comment by tsujan** [comment]
I don't think this is important but "military time" is just a name. As [Wikipedia](https://en.wikipedia.org/wiki/Date_and_time_notation_in_the_United_States) says, "The 24-hour notation is also widely used by astronomers, hospitals, various forms of transportation, and at radio and..." Nowadays, doctors, nurses and especially astronomers appreciate LXQt ;)

**Comment by shirishag75** [comment]
I just tried lxqt-desktop for the first time and am entirely comfortable with the 24 hour clock, i.e. the way it reads. The best of the both would be a way of setting so that users who love 12 hour clocks should be able to use them while those like me/us who love the 24 hour clocks because it's just easier time to remember or share timings that way

**Comment by isf63** [comment]
This is because of the advanced formatting used which is not locale-dynamic, I also think that the Short and Long formats also use [format strings](https://doc.qt.io/qt-6/qdatetime.html#toString) which are hardcoded. It would need to have added locale-aware format strings, which would be tricky without a standard database (that Qt may not provide.)
